### PR TITLE
Sameerch/keyvault

### PR DESCRIPTION
**Pull request by schabungbam**
The Azure Key Vault [Sample] connector.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by dmkuby** [comment]
Do we need an icon in public repo?

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,900 @@
...
+            "in": "query",
+            "name": "api-version",
+            "required": true,
+            "type": "string",
+            "x-ms-visibility": "internal"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": false,
```
Comment (dmkuby) [inline samples/AzureKeyVault/apiDefinition.swagger.json]: "value" field is required, but the parameter itself is optional. It means that empty payload is a valid request.
Is it expected or the parameter should be also required?

File: samples/AzureKeyVault/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "apiDefinition": "apiDefinition.swagger.json",
```
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/settings.json]: It makes things easier. :)

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
+                          "created": {
+                            "description": "created",
+                            "format": "int32",
+                            "type": "integer"
+                          },
+                          "enabled": {
+                            "description": "enabled",
+                            "type": "boolean"
+                          },
+                          "recoverylevel": {
```
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: In general, yes.  But this is generated by the portal - so I will skip this.

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,900 @@
...
+                      "type": "integer"
+                    }
+                  },
+                  "type": "object"
+                },
+                "key": {
+                  "description": "key",
+                  "properties": {
+                    "e": {
+                      "description": "e",
```
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Good point.  But given that it is just a sample, I will not fix this.  I added links to the API documentation for each method instead.

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,900 @@
...
+                      "encrypt",
+                      "decrypt",
+                      "sign",
+                      "verify",
+                      "wrapKey",
+                      "unwrapKey"
+                    ],
+                    "title": "",
+                    "type": "string",
+                    "x-ms-visibility": ""
```
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Good point. Fixed.  Looks like we have a bug in the portal.

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,900 @@
...
+                    "description": "JSON web key operations",
+                    "enum": [
+                      "encrypt",
+                      "decrypt",
+                      "sign",
+                      "verify",
+                      "wrapKey",
+                      "unwrapKey"
+                    ],
+                    "title": "",
```
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Fixed.

[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Portal bug.

File: samples/AzureKeyVault/apiDefinition.swagger.json (hunk 1 above)
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Good observation. I made body required as well.

**Comment by schabungbam** [comment]
> Do we need an icon in public repo?

Not required.

File: samples/AzureKeyVault/apiDefinition.swagger.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (dmkuby) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: Yes, I pointed this bug before, not sure if it was located and fixed or not.

File: samples/AzureKeyVault/settings.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShyamSunder90) [reply samples/AzureKeyVault/settings.json]: We are not checking in this file or the icon file in open source repro. lets be consistent

File: samples/AzureKeyVault/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,872 @@
...
+            "name": "body",
+            "required": true,
+            "schema": {
+              "properties": {
+                "attributes": {
+                  "description": "attributes",
+                  "properties": {
+                    "enabled": {
+                      "description": "Determines whether the object is enabled.",
+                      "enum": [
```
Comment (dmkuby) [inline samples/AzureKeyVault/apiDefinition.swagger.json]: This is another bug in the portal: type is Boolean, but enums are string. Please remove enums, it also should not pass swagger validation also.

File: samples/AzureKeyVault/settings.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (schabungbam) [reply samples/AzureKeyVault/settings.json]: why is that?

File: samples/AzureKeyVault/apiDefinition.swagger.json (hunk 7 above)
Comment (schabungbam) [reply samples/AzureKeyVault/apiDefinition.swagger.json]: We should file these bugs to be fixed.

**Pull request merged**